src/Zip.Shared/ZipEntry.cs

```diff
@@ -2422,7 +2429,76 @@ private static ZipEntry Create(string nameInArchive, ZipEntrySource source, Obje
         }
 
 
+        public ZipEntry CloneForNewZipFile(ZipFile newZipFile)
+        {
+            if (this.Source != ZipEntrySource.ZipFile)
+            {
+                throw new InvalidOperationException("The entry you are trying to add wasn't loaded from a zip file.");
+            }
+
+            if (this.IsChanged)
+            {
+                throw new InvalidOperationException("The entry you are trying to add was modified.");
+            }
 
+            if (this.IsDirectory)
+            {
+                throw new InvalidOperationException("The entry you are trying to add is a directory.");
+            }
+
+            if (!this.CompressionMethod.Equals(newZipFile.CompressionMethod))
```

Comment:
Could you please add a check like this one for _Encryption and _Password?

Reply:
I had the check on Encryption and Password, but had to remove them as they are not set on the zip2.

As I checked encryption can be mixed and as I don't decrypt I don't need the password.

Reply:
> As I checked encryption can be mixed and as I don't decrypt I don't need the password.
Ah, fair enough, just checked an it is indeed possible to have both encrypted and non-encrypted entries in the same zip, I guess it might be possible to have different passwords for them too.